### PR TITLE
Initial version of libteec and libckteec implementation

### PR DESCRIPTION
**Pull request by dsemenets**

### CODE REVIEW

**Comment by lorc** [comment]
There are checkpatch warnings and errors: https://buildbot.lorx.dev/#/builders/6/builds/3/steps/3/logs/stdio

**Comment by dsemenets** [comment]
That code taken from optee repo and I kept original code stile

**Comment by lorc** [comment]
Yes, but not all the warnings are about the code. For example:

```
76117c6f572f1ecc21b298df7a09e0def4258d72:69: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#69: FILE: Kconfig:6:
+config OPTEE_CLIENT
76117c6f572f1ecc21b298df7a09e0def4258d72:76: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#76: FILE: Kconfig:13:
+config OPTEE_LIBTEEC
76117c6f572f1ecc21b298df7a09e0def4258d72:81: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#81: FILE: Kconfig:18:
+config OPTEE_BENCHMARK
76117c6f572f1ecc21b298df7a09e0def4258d72:87: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#87: FILE: Kconfig:24:
+config OPTEE_DRV_STUB
```

**Comment by dsemenets** [comment]
@lorc Can you review This PR?

**Comment by lorc** [comment]
Could you please fix the checkpatch issues first? There are still errors that are not related to the imported code.

**Comment by lorc** [comment]
```
--------------------------------------------------------------------------
Commit 18433d5be82c ("libteec: Initial version of libteec implementation")
--------------------------------------------------------------------------
18433d5be82c3191ab2196ac48a54bd599268fcd:69: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#69: FILE: Kconfig:6:
+config OPTEE_CLIENT
18433d5be82c3191ab2196ac48a54bd599268fcd:77: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#77: FILE: Kconfig:14:
+config OPTEE_LIBTEEC
18433d5be82c3191ab2196ac48a54bd599268fcd:82: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#82: FILE: Kconfig:19:
+config OPTEE_BENCHMARK
18433d5be82c3191ab2196ac48a54bd599268fcd:88: WARNING:CONFIG_DESCRIPTION: please write a paragraph that describes the config symbol fully
#88: FILE: Kconfig:25:
+config OPTEE_DRV_STUB
```

**Comment by xakep-amatop** [comment]
both commits have the same descriptions is it right ?

**Comment by dsemenets** [comment]
Code for both commits was taken from the same repo.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023 EPAM Systems
+# Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
```
Comment (lorc) [inline CMakeLists.txt]: Well, thanks of course, but I am not author here :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+if (CONFIG_OPTEE_CLIENT)
```
Comment (lorc) [inline CMakeLists.txt]: There is `zephyr_include_directories_ifdef`.

File: libteec/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+add_library(OPTEE_LIBTEEC INTERFACE)
+
+target_include_directories(OPTEE_LIBTEEC INTERFACE include)
+
+zephyr_library()
+zephyr_library_sources(src/tee_client_api.c)
+zephyr_library_sources(src/teec_trace.c)
+zephyr_library_link_libraries(OPTEE_LIBTEEC)
+zephyr_library_sources_ifdef(CONFIG_OPTEE_DRV_STUB src/test_tee.c)
```
Comment (lorc) [inline libteec/CMakeLists.txt]: This commit  does not include this file. Please remove this line.

**Review by lorc — CHANGES_REQUESTED** [review]
> Code was taken from the optee repo:
https://github.com/OP-TEE/optee_client.git
revision: master
Original code style was kept.

Thank you for mentioning revision, but this is not entirely helpful, because tomorrow `master` will point to another commit. Could you please provide exact commit id used?

File: CMakeLists.txt (hunk 1 above)
Comment (dsemenets) [reply CMakeLists.txt]: Removed

File: CMakeLists.txt (hunk 2 above)
Comment (dsemenets) [reply CMakeLists.txt]: Fixed

File: libteec/CMakeLists.txt (hunk 3 above)
Comment (dsemenets) [reply libteec/CMakeLists.txt]: Done

**Review by lorc — DISMISSED** [review]
Thank you

`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by dsemenets** [comment]
@lorc Thanks. Can you merge this PR to master branch?

**Review by lorc — APPROVED** [review]

**Pull request merged**